nixos/modules/security/pam.nix

```diff
@@ -470,90 +558,108 @@ let
       setLoginUid = mkDefault cfg.startSession;
       limits = mkDefault config.security.pam.loginLimits;
 
+      text = let
+        # Formats a string for use in `module-arguments`. See `man pam.conf`.
+        formatModuleArgument = token:
+          if hasInfix " " token
+          then "[${replaceStrings ["]"] ["\\]"] token}]"
+          else token;
+        formatRules = type: pipe cfg.types.${type}.rules [
+          attrValues
+          (filter (rule: rule.enable))
+          (sort (a: b: a.order < b.order))
+          (map (rule: concatStringsSep " " (
+            [ type rule.control rule.modulePath ]
+            ++ map formatModuleArgument rule.args
+          )))
```

Comment:
It would be really nice to render the `order` of each rule above itself, so that one can just look at `/etc/pam.d/*` to see the values.

Reply:
Added `order` comments in 4995e085152bde3719f01affef1c5a1903d483bf.